PathTracking/rear_wheel_feedback/rear_wheel_feedback.py

```diff
@@ -147,64 +160,46 @@ def closed_loop_prediction(cx, cy, cyaw, ck, speed_profile, goal):
         v.append(state.v)
         t.append(time)
 
-        if target_ind % 1 == 0 and show_animation:
+        if show_animation:
             plt.cla()
             # for stopping simulation with the esc key.
             plt.gcf().canvas.mpl_connect('key_release_event',
                     lambda event: [exit(0) if event.key == 'escape' else None])
-            plt.plot(cx, cy, "-r", label="course")
+            plt.plot(track.X(s), track.Y(s), "-r", label="course")
             plt.plot(x, y, "ob", label="trajectory")
-            plt.plot(cx[target_ind], cy[target_ind], "xg", label="target")
+            plt.plot(track.X(s0), track.Y(s0), "xg", label="target")
             plt.axis("equal")
             plt.grid(True)
-            plt.title("speed[km/h]:" + str(round(state.v * 3.6, 2)) +
-                      ",target index:" + str(target_ind))
+            plt.title("speed[km/h]:{:.2f}, target s-param:{:.2f}".format(round(state.v * 3.6, 2), s0))
             plt.pause(0.0001)
 
     return t, x, y, yaw, v, goal_flag
 
+def calc_target_speed(state, yaw_r):
+    target_speed = 10.0 / 3.6
 
-def calc_speed_profile(cx, cy, cyaw, target_speed):
-
-    speed_profile = [target_speed] * len(cx)
-
-    direction = 1.0
-
-    # Set stop point
-    for i in range(len(cx) - 1):
-        dyaw = cyaw[i + 1] - cyaw[i]
-        switch = math.pi / 4.0 <= dyaw < math.pi / 2.0
-
-        if switch:
-            direction *= -1
-
-        if direction != 1.0:
-            speed_profile[i] = - target_speed
-        else:
-            speed_profile[i] = target_speed
-
-        if switch:
-            speed_profile[i] = 0.0
-
-    speed_profile[-1] = 0.0
+    dyaw = yaw_r - state.yaw
+    switch = math.pi / 4.0 <= dyaw < math.pi / 2.0
 
-    return speed_profile
+    if switch:
+        state.direction *= -1
+        return 0.0
+    
+    if state.direction != 1:
+        return -target_speed
 
+    return target_speed
 
 def main():
     print("rear wheel feedback tracking start!!")
     ax = [0.0, 6.0, 12.5, 5.0, 7.5, 3.0, -1.0]
     ay = [0.0, 0.0, 5.0, 6.5, 3.0, 5.0, -2.0]
     goal = [ax[-1], ay[-1]]
 
-    cx, cy, cyaw, ck, s = cubic_spline_planner.calc_spline_course(
-        ax, ay, ds=0.1)
-    target_speed = 10.0 / 3.6
-
-    sp = calc_speed_profile(cx, cy, cyaw, target_speed)
+    track = CubicSplinePath(ax, ay)
```

Comment:
track -> cubic_spline_path ?

Reply:
couldn't this object be better to have the name detached from the interpolation method?

Reply:
Does this object means a spline path object? What do you imply the "track" name?

Reply:
I meant the name could be any of reference_track, reference_path, target_track and target_path or even only path or track, but it isn't necessary to be cubic spline as long as class C2.

Reply:
oh ok. I got it. i feel reference_path is better.

Reply:
changed.